errors/types.go

```diff
@@ -127,6 +127,20 @@ func IsDownloadDeclinedError(err error) bool {
 	return false
 }
 
+// IsDownloadDeclinedError indicates that the upload operation failed because of
+// an HTTP 422 response code.
+func IsUnprocessableEntityError(err error) bool {
+	if e, ok := err.(interface {
```

Comment:
I've never seen this syntax before! Could you explain what it's doing here?

Reply:
Definitely! This is a typecast, but it looks a little odd since the type is defined inline. Normally, you'd see something like:

```go
func doSomethingIfReader(v interface{}) {
        if r, ok := v.(io.Reader); ok {
                // ok is 'true' because 'v' is an 'io.Reader' underneath.
                // So, we can use any function that io.Reader receives,
                // or use 'r' as a bona-fide io.Reader.
                io.Copy(os.Stdout, r)
        }
}

But, inside the `()` (where `io.Reader` is above), you can define any type, including doing so inline.

So, this example defines an `interface` with the function above, and then checks to see if `err` implements that (i.e., is a type that receives all of those methods). If it is, `e` receives those methods, and `ok` is true. If not, `ok` is false.

Particularly, we're checking if either `err` or any of its parents are this sentinel type by checking if it receives a method. If so, we return true, and if not, we return false otherwise. In practice, this isn't a super common technique, but it is the pattern in this file, so I'm inclined to go with that over something else.